src/component.js

```diff
@@ -395,4 +390,19 @@ export default class Component {
             }
         })
     }
+
+    executeAndClearNextTickStack() {
+        const self = this
+
+        const later = function () {
+            this.nextTickTimeout = null
+            while (self.nextTickStack.length > 0) {
+                self.nextTickStack.shift()()
+            }
+        }
+
+        clearTimeout(this.nextTickTimeout)
+
+        this.nextTickTimeout = setTimeout(later, 10)
```

Comment:
I don't like adding an arbitrary buffer of 10ms. I'd much rather an implementation that is truly the "nextTick" meaning (which is usually achieved by a setTimeout of 0)

Does that make sense?

I'm grateful you found the bug and contributed a test, just hoping for a better implementation.

Reply:
yeah, there was some sort of race condition between the 2 debounce but i'm not a big fan of the duplicated debounce. I'll see if i can fix it in a different way. Thanks for reviewing

Reply:
Got a simpler solution which does not involve debouncing and adding a timeout.